qiskit_optimization/converters/linear_inequality_to_penalty.py

```diff
@@ -0,0 +1,335 @@
+            for (j, z) in enumerate(rowlist):
+                for k in range(j, len(rowlist)):
+                    # if j and k already exist in the quadratic terms dict,
+                    # add a penalty term into existing value
+                    # else create new key and value in the quadratic term dict
+
+                    if conv_matrix[j + 1][k + 1] != 0:
+                        tup = cast(Union[Tuple[int, int], Tuple[str, str]], (z[0], rowlist[k][0]))
+                        quadratic[tup] = (
+                            quadratic.get(tup, 0.0) + sense * penalty * conv_matrix[j + 1][k + 1]
+                        )
+
+        # Copy quadratic_constraints
+        for quadratic_constraint in problem.quadratic_constraints:
+            self._dst.quadratic_constraints.append(quadratic_constraint)
+
+        if problem.objective.sense == QuadraticObjective.Sense.MINIMIZE:
+            self._dst.minimize(offset, linear, quadratic)
+        else:
+            self._dst.maximize(offset, linear, quadratic)
+
+        # Update the penalty to the one just used
+        self._penalty = penalty  # type: float
+
+        return self._dst
+
+    def _conversion_matrix(self, constraint) -> np.ndarray:
+        """Construct conversion matrix for special constraint.
+
+        Returns:
+            Return conversion matrix which is used to construct
+            penalty term in main function.
+
+        """
+
+        vars_dict = constraint.linear.to_dict(use_name=True)
+        vrs = list(vars_dict.items())
+        rhs = constraint.rhs
+        sense = constraint.sense
+
+        num_vars = len(vrs)
+        combinations = list(itertools.combinations(np.arange(num_vars), 2))
+
+        # conversion matrix
+        conv_matrix = np.zeros((num_vars + 1, num_vars + 1), dtype=int)
+
+        for combination in combinations:
+            index1 = combination[0] + 1
+            index2 = combination[1] + 1
+
+            if rhs in (1, num_vars - 1):
+                if sense == ConstraintSense.GE:
+                    conv_matrix[0][0] = conv_matrix[0][0] + 1
+                    conv_matrix[0][index1] = conv_matrix[0][index1] - 1
+                    conv_matrix[0][index2] = conv_matrix[0][index2] - 1
+                conv_matrix[index1][index2] = 1
+            elif rhs == 0:
+                if sense == ConstraintSense.GE:
+                    if vrs[index1 - 1][1] < 0.0:
+                        conv_matrix[0][index1] = 1
+                    elif vrs[index2 - 1][1] < 0.0:
+                        conv_matrix[0][index2] = 1
+                elif sense == ConstraintSense.LE:
+                    if vrs[index1 - 1][1] > 0.0:
+                        conv_matrix[0][index1] = 1
+                    elif vrs[index2 - 1][1] > 0.0:
+                        conv_matrix[0][index2] = 1
+
+                conv_matrix[index1][index2] = -1
+
+        return conv_matrix
+
+    def _is_special_constraint(self, problem, constraint) -> bool:
```

Comment:
"_is_target_constraint" might be good as mentioned https://github.com/Qiskit/qiskit-optimization/pull/157/files#r650786873

Reply:
Changed to `_is_matched_constraint`